hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/compress/snappy/SnappyCompressor.java

```diff
@@ -234,8 +234,6 @@ public int compress(byte[] b, int off, int len)
       finished = true;
     }
 
-    // Get atmost 'len' bytes
-    n = Math.min(n, len);
```

Comment:
Do we need this change? This could cause IndexOutOfBoundsException on [consuming compressed data](https://github.com/apache/hadoop/blob/trunk/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/compress/CompressorStream.java#L91-L93).

Reply:
This change is not needed. Reverted the change.

I would like to add a test case for this problem.